src/arm7tdmi.rs

```diff
@@ -86,12 +92,45 @@ impl Arm7tdmi {
     fn branch_link(&mut self, op_code: u32) {
         todo!("Branch Link")
     }
+
+    fn mov(&mut self, op_code: u32) {
+        // bits [24-21] are the RD
+        let rd = (op_code & 0x00_00_F0_00) >> 12;
+        println!("RD: {:?}", rd);
+
+        // 25th bit is I = Immediate Flag
+        let immediate: bool = (op_code & 0x02_00_00_00) >> 25 == 1;
+        println!("Immediate: {:?}", immediate);
+
+        // 20th bit is S = Condition Set
+        if op_code & 0x00_08_00_00 > 0 {
+            todo!("Condition set")
+        }
+
+        if immediate {
+            // bits [7-0] are the immediate value
+            let immediate_value = op_code & 0x00_00_00_FF;
+            println!("value: {:?}", immediate_value);
+
+            // the instruction is MOV RD, immediate_value
+            self.registers[rd as usize] = immediate_value;
+        } else {
+            todo!("Not implemented yet.");
+        }
+
+        self.program_counter += 4;
```

Comment:
I think that this is bad. If an instruction is `MOV PC, offset` it will be overwritten. I'll add a test to check also this behavior.

Reply:
Not sure about it. I left a comment in the code.